hive_agent/tools/retriever/base_retrieve.py

```
@@ -0,0 +1,109 @@
+from llama_index.core import VectorStoreIndex, SimpleDirectoryReader
+from hive_agent.server.routes import files
+
+supported_exts = [".md", ".mdx", ".txt", ".csv", ".docx", ".pdf"]
+
+
+class RetrieverBase:
+    def __init__(
+        self,
+        required_exts=supported_exts,
+        retrieve_data_path=files.BASE_DIR,
+        name="BaseRetriever",
+        description="This tool creates a base retriever index",
+    ):
+        self.retrieve_data_path = retrieve_data_path
+        self.required_exts = required_exts
+        self.name = name
+        self.description = description
+
+    def _load_documents(self, file_path=None, folder_path=None):
+        if file_path is None:
+            if folder_path is None:
+                folder_path = self.retrieve_data_path
+
+        reader = SimpleDirectoryReader(
+            input_files=file_path,
+            input_dir=folder_path,
+            required_exts=self.required_exts,
+            recursive=True,
+            filename_as_id=True,
+        )
+        return reader.load_data()
+
+    def create_basic_index(self, file_path=None, folder_path=None):
+        documents = self._load_documents(file_path, folder_path)
+        index = VectorStoreIndex.from_documents(documents)
+        return index
+
+    def insert_documents(self, index, file_path=None, folder_path=None):
+        documents = self._load_documents(file_path, folder_path)
+        if not documents:
+            return "No documents found to insert."
```

should we raise an error with the error message here instead?